File: backend/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
namespace Ecommerce.Models;

public class Product : BaseModel
{
    [MaxLength(256)]
    public string Title { get; set; } = string.Empty;
    public int Price { get; set; }
    public string? Description { get; set; }
    public string? Image { get; set; }
    [NotMapped]
    public Category Category { get; set; } = null!;
    public int CategoryId { get; set; }
    [NotMapped]
    [JsonIgnore]
    public Order Order { get; set; } = null!;
    public int OrderId { get; set; }
}

File: frontend/src/types/product.ts
import { Category } from "./category";

export interface Product {
  id: number;
  title: string;
  price: number;
  description: string;
  category: Category;
  image: string;
  filterId: null;
}

export interface CreateProductType {
  title: string;
  description: string;
  price: number;
  categoryId: number;
  image: string;
  orderId: number;
}
export interface UpdateProduct {
  title: string;
  description: string;
  price: number;
  categoryId: number | undefined;
  image: string | undefined;
  orderId: number;
  id: number;
}
export type ProductId = number;
